packages/xstate-solid/src/useInterpret.ts

```diff
@@ -0,0 +1,127 @@
+import type {
+  AnyStateMachine,
+  InterpreterOptions,
+  Observer,
+  AreAllImplementationsAssumedToBeProvided,
+  InternalMachineOptions,
+  InterpreterFrom
+} from 'xstate';
+import { State, interpret, toObserver } from 'xstate';
+import type { UseMachineOptions, MaybeLazy } from './types';
+import { onCleanup, onMount } from 'solid-js';
+
+type RestParams<
+  TMachine extends AnyStateMachine
+> = AreAllImplementationsAssumedToBeProvided<
+  TMachine['__TResolvedTypesMeta']
+> extends false
+  ? [
+      options: InterpreterOptions &
+        UseMachineOptions<TMachine['__TContext'], TMachine['__TEvent']> &
+        InternalMachineOptions<
+          TMachine['__TContext'],
+          TMachine['__TEvent'],
+          TMachine['__TResolvedTypesMeta'],
+          true
+        >,
+      observerOrListener?:
+        | Observer<
+            State<
+              TMachine['__TContext'],
+              TMachine['__TEvent'],
+              any,
+              TMachine['__TTypestate'],
+              TMachine['__TResolvedTypesMeta']
+            >
+          >
+        | ((
+            value: State<
+              TMachine['__TContext'],
+              TMachine['__TEvent'],
+              any,
+              TMachine['__TTypestate'],
+              TMachine['__TResolvedTypesMeta']
+            >
+          ) => void)
+    ]
+  : [
+      options?: InterpreterOptions &
+        UseMachineOptions<TMachine['__TContext'], TMachine['__TEvent']> &
+        InternalMachineOptions<
+          TMachine['__TContext'],
+          TMachine['__TEvent'],
+          TMachine['__TResolvedTypesMeta']
+        >,
+      observerOrListener?:
+        | Observer<
+            State<
+              TMachine['__TContext'],
+              TMachine['__TEvent'],
+              any,
+              TMachine['__TTypestate'],
+              TMachine['__TResolvedTypesMeta']
+            >
+          >
+        | ((
+            value: State<
+              TMachine['__TContext'],
+              TMachine['__TEvent'],
+              any,
+              TMachine['__TTypestate'],
+              TMachine['__TResolvedTypesMeta']
+            >
+          ) => void)
+    ];
+
+export function useInterpret<TMachine extends AnyStateMachine>(
```

Comment:
More of a helper and API parity. There is a way to use it in Solid, but you don't get fine grained updates. The [from](https://www.solidjs.com/docs/latest/api#from) function turns any observable into a signal. It would trigger an update on anything consuming it though, so not fine grained

Reply:
Ok, so this could be usable~ but actually should be discouraged, right? If it should be discouraged then I would prefer keeping this outside of the library so people don't shoot themselves in the foot accidentally. I guess though that this depends on the final API that we end up with (I'm gonna try to make a decision on this in the upcoming week) - because we might need `createService`, just with somewhat different semantics 🤔

Reply:
Yeah, this is usable but likely unneeded as it doesn't add any reactivity in its current state. The current hook is more of a launching off point to other hooks. Using `from` would be good enough for basic reactivity. Here is how you would get a usable signal: 
```typescript
const service = interpret(machine).start();
const serviceState = from(service); // Automatically subscribes and triggers updates to serviceState signal - not deeply reactive
```